cognite/client/_api_client.py

```diff
@@ -79,7 +79,7 @@ def __init__(
         self._config = config
         self._api_version = api_version
         self._api_subversion = config.api_subversion
-        self._cognite_client = cognite_client
+        self._cognite_client = cast("CogniteClient", cognite_client)
```

Comment:
Not make it Optional instead? Do we ever not pass it?

Reply:
Agreed, seems like a much better solution. If this causes a lot of `mypy`-cascading issues, I'll put it in a separate PR.

Reply:
...yup, definitely separate PR 😂 

![image](https://user-images.githubusercontent.com/8521241/226344442-f2dedd83-5806-4dad-a23a-1bb83df2aecd.png)

Reply:
Addressed in this PR: https://github.com/cognitedata/cognite-sdk-python/pull/1175